### PR TITLE
Test/post branch

### PR DESCRIPTION
**Pull request by xynazog**
This PR addresses the ticket of creating a wishlist and creating wishlist items ( #57 )
The only changes been made are to test_wishlists.py file.

### CODE REVIEW

**Review by pinocchiochang — COMMENTED** [review]
Seems I just merge Palak's pull request, will this has conflicts with hers?

**Review by pinocchiochang — APPROVED** [review]
Good! Good!

File: test_wishlists.py [hunk 1]
```diff
@@ -19,6 +19,85 @@ def setUp(self):
...
+	def test_read_wishlist_item_wishlist_not_found(self):
+		resp = self.app.get('/wishlists/wl2/items/item2')
+		self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+	"""
+		Working test case.
+		This is a test case to check whether a wishlist is created or not.
+		POST verb checked here.
+	"""
+	def test_create_wishlist(self):
```
Comment (palakbhasin) [inline test_wishlists.py:28]: You should check that the number of wishlists increases and that the location header is returned.

File: test_wishlists.py [hunk 2]
```diff
@@ -19,6 +19,85 @@ def setUp(self):
...
+		self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+		new_json = json.loads(resp.data)
+		self.assertEqual(new_json['name'],'xynazog')
+
+	"""
+		Working test case.
+		This is a test case to check whether an item is added to a wishlist or not.
+		POST verb is checked here.
+	"""
+	def test_create_wishlist_item(self):
```
Comment (palakbhasin) [inline test_wishlists.py:44]: Theres nothing wrong with it but we already have a test wishlist so there's no need to create a new one. Also, you should check if the number of items increases and that the location header is returned

File: test_wishlists.py [hunk 3]
```diff
@@ -19,6 +19,85 @@ def setUp(self):
...
+	"""
+		Not working test case.
+		This is a test case to check whether an item is added to a wishlist out of index.
+		POST verb is checked here.
+	"""
+	def test_create_wishlist_item_wishlist_not_found(self):
+		new_item = {'id':'item3','description':'test item 3'}
+		data = json.dumps(new_item)
+		resp = self.app.post('/wishlists/3/items',data=data,content_type='application/json')
+		self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
```
Comment (palakbhasin) [inline test_wishlists.py]: The status code should be HTTP_404_NOT_FOUND, why are you checking for created?

File: test_wishlists.py (hunk 2 above)
Comment (palakbhasin) [reply test_wishlists.py:44]: Also, I think you should validate data in wishlists.py

**Review by palakbhasin — CHANGES_REQUESTED** [review]

File: test_wishlists.py (hunk 3 above)
Comment (xynazog) [reply test_wishlists.py]: Done! It should be 400_Bad_Request.

Comment (palakbhasin) [reply test_wishlists.py]: Are you sure? If the wishlist doesn't exist shouldnt it be 404? Isn't 400 for when the data isnt valid?

File: test_wishlists.py (hunk 1 above)
Comment (xynazog) [reply test_wishlists.py:28]: Done about number of wishlists increasing to two.

File: test_wishlists.py (hunk 2 above)
Comment (xynazog) [reply test_wishlists.py:44]: I was just checking the number of wishlists created. No need to create one! Sorry.

Comment (xynazog) [reply test_wishlists.py:44]: Can you tell me more about validation? I mean what do I have to validate? @palakbhasin

Comment (palakbhasin) [reply test_wishlists.py:44]: In our original wishlists.py we don't check if the data is valid. In my pr I added one to wishlists.py, so I think you should use that is_valid method for creating wishlists and items

Comment (pinocchiochang) [reply test_wishlists.py:44]: In our tests, should we also use your is_valid method? @palakbhasin

Comment (xynazog) [reply test_wishlists.py:44]: Can you check now @palakbhasin ? I guess it should be fine.

File: wishlists.py [hunk 4]
```diff
@@ -27,35 +27,43 @@ def index():
...
+    """
+    data = request.get_json()
+    if is_valid(data,'item'):
+        try:
+            tempDic = {}
+            tempDic['id'] = request.json['id']
+            tempDic['description'] = request.json['description']
+            return db.add_item(wishlist_id,tempDic), HTTP_201_CREATED
+        except WishlistException:
+            return jsonify(message='Cannot add a new item %s' % request.json['id']), HTTP_400_BAD_REQUEST
```
Comment (palakbhasin) [inline wishlists.py:63]: I think this should be:

Comment (palakbhasin) [reply wishlists.py:63]: `        except WishlistException:
            message = { 'error' : 'Wishlist %s was not found' % wishlist_id }
            return jsonify(message), HTTP_404_NOT_FOUND`

**Pull request merged**